src/features/vod/reducers/vodCategories.reducer.tsx

```diff
@@ -0,0 +1,70 @@
+import { createSlice } from '@reduxjs/toolkit';
+import {
+  RootState,
+  VODCategoriesState,
+  VODCategory,
+  VODSubCategory,
+  VODVideo,
+} from '../../../constants/types';
+import { store } from '../../../constants/store';
+
+const initialState: VODCategoriesState = {
+  categories: [],
+  loading: true,
+};
+
+const categoriesSlice = createSlice({
+  name: 'vod-categories',
+  initialState,
+  reducers: {
+    setVODCategories: (state, action) => {
+      state.categories = action.payload.categories;
+      state.loading = false;
+    },
+    setChosenVideo: (state, action) => {
+      state.chosenVideo = action.payload.video;
+    },
+    setChosenCategory: (state, action) => {
+      state.chosenCategory = {
+        category: action.payload.category,
+        subCategory: action.payload.subCategory,
+      };
+    },
+  },
+});
+
+export const setVODCategories = (categories: VODCategory[] | null) => {
+  store.dispatch(
+    categoriesSlice.actions.setVODCategories({
+      categories: categories,
+    }),
+  );
+};
+
+export const setChosenVideo = (chosenVideo: VODVideo) => {
```

Comment:
This is redundant!
you can export all these function at the end of the file with:
export const { setVODCategories, setChosenVideo, setChosenCategory } = categoriesSlice.actions;